src/lib.rs

```diff
@@ -12,6 +12,16 @@
 
 #![allow(dead_code)]
 
+// Even when trace logging is disabled, the trace macro has a significant
+// performance cost so we disable it in release builds.
+macro_rules! trace {
+    ($($tt:tt)*) => {
+        if cfg!(debug_assertions) {
+            ::log::trace!($($tt)*);
+        }
```

Comment:
Doesn't log have a feature to specify the minimum log level at compile time and then compile out every log below this level?

Reply:
It does but:
- it's disabled by default.
- this also disables uses of `trace!` outside of regalloc2.

I personally think that regalloc2's use of trace is just way too verbose to be enabled in release builds.

Reply:
Trace is meant for things that are extremely verbose. I think trace should almost always be disabled in release builds. You can use the `release_max_level_debug` feature of log for that.

https://docs.rs/log/latest/log/enum.Level.html

> Trace
> 
> The "trace" level.
>
> Designates very low priority, often extremely verbose, information.

Reply:
If I have a 100k functions to compile then logging ~10 lines per function is already very verbose. This is what I am currently using `trace` for. Having regalloc2 spew out thousands of lines per function is just unreasonable.

Reply:
I guess so.

Reply:
The `trace!()` uses in this crate are more-or-less "stuff I needed to printf-debug" and are definitely verbose; I find it to be the best way to debug very complex algorithmic code, and leave it in so I can get it back when encountering a new fuzzbug or whatever, but it definitely doesn't have to pollute trace-level output globally.

Maybe we could put this under a specific Cargo feature (`detailed-trace` or somesuch) that's off by default? That would keep me happy (rebuild with feature when debugging) while removing this from users' trace output.